### ISSUE TITLE
Using PrettyTime on Android - null pointer exception

### ISSUE THREAD

**Issue by jsarson** [MenoData/Time4J#316]
Hi, 

I found this library couple days ago and it is just amazing! 

Unfortunately I have some troubles with using PrettyTime on Android. When I run demo example, the app crash on second line during processing print method and throw null pointer exception.

Thank you for any help!

I'm using gradle with current time4j setup:
```gradle
compile 'net.time4j:time4j-android:3.3'
compile 'net.time4j:time4j-core:3.3'
compile 'net.time4j:time4j-i18n:3.3'
compile 'net.time4j:time4j-range:3.3'
```

Testing code:
```java
Duration<?> dur = Duration.of(337540, ClockUnit.SECONDS).with(Duration.STD_CLOCK_PERIOD);
String formatedDuration = PrettyTime.of(Locale.FRANCE).print(dur, TextWidth.WIDE);
```
Exception:
```java
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.util.Locale.equals(java.lang.Object)' on a null object reference
            at net.time4j.i18n.UnitPatternProviderSPI.getPattern(UnitPatternProviderSPI.java:379)
            at net.time4j.i18n.UnitPatternProviderSPI.getUnitPattern(UnitPatternProviderSPI.java:332)
            at net.time4j.i18n.UnitPatternProviderSPI.getYearPattern(UnitPatternProviderSPI.java:63)
            at net.time4j.UnitPatterns.lookup(UnitPatterns.java:377)
            at net.time4j.UnitPatterns.lookup(UnitPatterns.java:360)
            at net.time4j.UnitPatterns.<init>(UnitPatterns.java:132)
            at net.time4j.UnitPatterns.of(UnitPatterns.java:210)
            at net.time4j.PrettyTime.print(PrettyTime.java:605)
            at net.time4j.PrettyTime.format(PrettyTime.java:1076)
            at net.time4j.PrettyTime.print(PrettyTime.java:746)
            at net.time4j.PrettyTime.print(PrettyTime.java:653)
```

**Comment by MenoData** [comment]
Thank you for your feedback. The exception is surprising because in Java the code works fine. Android does seem to be different in context of resource bundles (I had experienced another problem related to resource bundles on Android in the past). Well, the NPE only tells that `ResourceBundle.getLocale()` yields `null` (although there is absolutely no hint in the javadoc of Android that this method is supposed to return no value at all - see http://developer.android.com/reference/java/util/ResourceBundle.html#getLocale%28%29).

Seems I have to dive deeper into Android source code and do extra tests in Android emulator.

**Comment by MenoData** [comment]
I could reproduce the problem on Android emulator and found out that the method `getLocale()` in Android-class `ResourceBundle` always produces `null` (within Time4J-configuration using a special resource-bundle-format). As workaround - the specialized `ResourceBundle`-subclass in Time4J will now keep an extra reference to the current locale candidate in use. Finally I have verified with this fix that your Android code will no longer throw an NPE but produce the expected output  "93 heures, 45 minutes et 40 secondes".

**Comment by jsarson** [comment]
Thank you for so quick fix! Is it possible to use this fix version via maven central?

**Comment by MenoData** [comment]
I have just now released v3.4. It is immediately available on GitHub and during the next few hours also on Maven Central (dependent on their administration time and job schedules). Note that I had to postpone the umalqura-calendar to next release v3.5 to release the fix now.

By the way, is there a special reason why you don't use the independent tz-repository of Time4J (leaving away the olson-module and tzdata-1.4-2015e)? I am just curious. With your configuration Time4J will just use a thin wrapper around `java.util.TimeZone`, effectively use the tz data of Android which might not be up-to-date on every device. If you are interested in that feature then please read the installation notes of the tutorial (http://time4j.net/tutorial/installation.html), especially the section about Android and packaging options.

**Comment by jsarson** [comment]
I appreciate all of your help. Now everything works like a charm on Android :-)

By the way, there is no special reason to don't use independent tz-repository - I just didn't find enought info about it. Now it is clear. Thanks
